pkg/i2gw/provider.go

```diff
@@ -105,3 +107,35 @@ type GatewayResources struct {
 // Different FeatureParsers will run in undetermined order. The function must
 // modify / create only the required fields of the gateway resources and nothing else.
 type FeatureParser func([]networkingv1.Ingress, *GatewayResources) field.ErrorList
+
+type ProviderSpecificConf struct {
+	Name         string
+	Description  string
+	DefaultValue string
+}
+
+var (
+	providerSpecificConfs      = map[ProviderName]map[string]ProviderSpecificConf{}
+	providerSpecificConfsMutex = sync.RWMutex{}
```

Comment:
Why mutex as a global var instead of wrapping `map[ProviderName]map[string]ProviderSpecificConf{}` as a new type and add the mutex within it?